rpc/src/rpc_service.rs

```diff
@@ -81,13 +76,7 @@ struct RpcRequestMiddleware {
     snapshot_config: Option<SnapshotConfig>,
     bank_forks: Arc<RwLock<BankForks>>,
     health: Arc<RpcHealth>,
-    cluster_info: Arc<ClusterInfo>,
-    block_commitment_cache: Arc<RwLock<BlockCommitmentCache>>,
-    vote_accounts_to_monitor: Arc<HashSet<Pubkey>>,
-    enable_prometheus_metrics: bool,
-    /// Initialized based on vote_accounts_to_monitor, maps identity
-    /// pubkey associated with the vote account to the validator info.
-    identity_info_map: Arc<IdentityInfoMap>,
+    prometheus_metrics: Option<Arc<PrometheusMetrics>>,
```

Comment:
:ok_hand: Nice simplification to pack this into an optional struct, much cleaner than the pile of arguments + bool.